### ISSUE TITLE
[stable/prometheus-operator] podMetadata annotations not being propagated to Prometheus instance pod

### ISSUE THREAD

**Issue by revog** [helm/charts#20503]
**Describe the bug**
I need to add an annotation to the running prometheus instance. I used the helm chart as follows (unnessecary information ommited):
```
prometheus:
  enabled: true
  prometheusSpec:
  ...
    podMetadata:
      annotations:
        backup.velero.io/backup-volumes: prometheus-prometheus-federated-tst-p-prometheus-db
  ...
```

Unfortunately this annotation is not set after creating the prometheus pod:
```
kubectl -n prometheus-federated-tst describe pod prometheus-prometheus-federated-tst-p-prometheus-0
Name:           prometheus-prometheus-federated-tst-p-prometheus-0
Namespace:      prometheus-federated-tst
Priority:       0
Node:           *********/**.**.**.**
Start Time:     Mon, 03 Feb 2020 13:36:23 +0100
Labels:         app=prometheus
                controller-revision-hash=prometheus-prometheus-federated-tst-p-prometheus-bdb9bd8f5
                prometheus=prometheus-federated-tst-p-prometheus
                statefulset.kubernetes.io/pod-name=prometheus-prometheus-federated-tst-p-prometheus-0
Annotations:    kubernetes.io/limit-ranger:
                  LimitRanger plugin set: cpu, memory request for container prometheus-config-reloader; cpu, memory limit for container prometheus-config-re...
                kubernetes.io/psp: prometheus-federated-tst-p-prometheus
Status:         Running
...
```

**Version of Helm and Kubernetes**:
Kubernetes 1.15
Helm  v2.12.0

**Which chart**:
Chart: stable/prometheus-operator 
Version:  8.6.0

**What happened**:
Annotation missing.

**What you expected to happen**:
Existing annotation on pod

**How to reproduce it** (as minimally and precisely as possible):

**Anything else we need to know**:
Regarding to issue #16272 it should be working.

**Comment by jbfavre** [comment]
I also face this bug.
It's introduced with either Chart 8.6.0 or prometheus-operator 0.35.0 or CRDs refacto (or a mix of all)

As a **workaround**: from Chart 8.5.15, bump chart version to 8.6.0, and set `.Values.prometheusOperator.manageCrds` to `false`.
This will prevent prometheus-operator to manage (and thus update) CRDs.

Downside is that CRDs aren't updated anymore.

I'm running Chart 8.7.0 with prometheus-operator 0.35.0 and CRDs from Chart 8.5.15 without problems, but it's not a suitable situation on long term.

**Comment by jbfavre** [comment]
Also note that this could be linked to https://github.com/helm/charts/issues/20544:
CRDs provided by Chart >=8.6.0 (commit f47e033003e45babf5375cf5437711298cb7ea2e from @vsliouniaev) have duplicate annotations

**Comment by vsliouniaev** [comment]
You must use `prometheus.podMetadata` and `alertmanager.podMetadata` to set information on pods created by the operator.

https://github.com/coreos/prometheus-operator/blob/master/Documentation/api.md#prometheusspec

**Comment by revog** [comment]
> You must use `prometheus.podMetadata` and `alertmanager.podMetadata` to set information on pods created by the operator.
> 
> https://github.com/coreos/prometheus-operator/blob/master/Documentation/api.md#prometheusspec

Hmm are you really sure? I've just tried adding the `podMetadata` directly below `prometheus:` but hat no luck.

According to the default [values.yaml](https://github.com/helm/charts/blob/master/stable/prometheus-operator/values.yaml) this `podMetadata` parameters are subs of `alertmanagerSpec` respectively `prometheusSpec`:

```
alertmanager:
...
  alertmanagerSpec:
    ## Standard object’s metadata. More info: https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#metadata
    ## Metadata Labels and Annotations gets propagated to the Alertmanager pods.
    ##
    podMetadata: {}
...
```
```
prometheus:
...
  prometheusSpec
    ## Standard object’s metadata. More info: https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#metadata
    ## Metadata Labels and Annotations gets propagated to the prometheus pods.
    ##
    podMetadata: {}
    # labels:
    #   app: prometheus
    #   k8s-app: prometheus
```

**Comment by jbfavre** [comment]
> You must use `prometheus.podMetadata` and `alertmanager.podMetadata` to set information on pods created by the operator.
> 
> https://github.com/coreos/prometheus-operator/blob/master/Documentation/api.md#prometheusspec

You mean `prometheus.prometheusSpec.podMetadata` and `alertmanager.alertmanagerSpec.podMetadata`, right ?
If so, I can say it doesn't work with latest Chart >=8.6.0

**Comment by FGSHadden** [comment]
I can confirm that his issue persists with 8.7 using `prometheus.prometheusSpec.podMetadata and alertmanager.alertmanagerSpec.podMetadata` setting `manageCrds` to false is a workaround to resolve the issue. I am already deploying the crds in a seperate workflow to get around the Helm 2.x issues, so its not much of a problem for me. None of the defined label values are being assingned to the alertmanager and operator pods when `prometheusOperator.manageCrds` is set to True.

**Comment by mogaal** [comment]
I am using `8.7.0` and for some reason, it does work for me:

```
prometheus:
  prometheusSpec:
    podMetadata:
      annotations:
        iam.amazonaws.com/role: "monitoring.aws.mogaal.com"
```

I can see the info the annotations:

```console
$ kubectl -n monitoring get pod prometheus-prometheus-operator-prometheus-0 -o yaml | head -n 8
apiVersion: v1
kind: Pod
metadata:
  annotations:
    iam.amazonaws.com/role: monitoring.mogaal.cloud-platform.service.justice.gov.uk
```

**Comment by mogaal** [comment]
I am using `8.7.0` and for some reason, it does work for me:

```
prometheus:
  prometheusSpec:
    podMetadata:
      annotations:
        iam.amazonaws.com/role: "monitoring.aws.mogaal.com"
```

I can see the right annotations:

```console
$ kubectl -n monitoring get pod prometheus-prometheus-operator-prometheus-0 -o yaml | head -n 8
apiVersion: v1
kind: Pod
metadata:
  annotations:
    iam.amazonaws.com/role: "monitoring.aws.mogaal.com"
```

**Comment by FGSHadden** [comment]
hi @mogaal 

The issue is specifically with the Labels: 

prometheus:
  prometheusSpec:
    podMetadata:
      Labels:

Labels are not being applied to Alertmanager and Operator Pods during install.

**Comment by jbfavre** [comment]
@mogaal thanks for the update. I suppose you're running prometheus-operator 0.35.0 ?
Could you also provide your version of Kubernetes, Helm and so on, please ?

**Comment by jbfavre** [comment]
For me:
* Helm v2.16.1
* k8s: v1.15.7-gke.23

**Comment by FGSHadden** [comment]
for me:

- Helm 2.16.1
- k8s: 1.16.6
- Prometheus-Operator 0.35.0

**Comment by jbfavre** [comment]
Just tested with Chart 8.10.3
Everything works now so far…

**Comment by jbfavre** [comment]
OK, now it's working.
Fix: Use operator 0.37.0 release which include a bug fix (https://github.com/coreos/prometheus-operator/issues/3041)

**Comment by BrianKopp** [comment]
Ditto, updated to `prometheus.image.tag: v0.38.1` and throat punched the metadata key. Metadata restored!
